pkg/conn/baseconn.go

```diff
@@ -56,14 +64,31 @@ type BaseConn struct {
 	DBConn *sql.Conn
 
 	RetryStrategy retry.Strategy
+
+	// for reset
+	ParentDB *BaseDB
 }
 
 // newBaseConn builds BaseConn to connect real DB
-func newBaseConn(conn *sql.Conn, strategy retry.Strategy) *BaseConn {
+func newBaseConn(conn *sql.Conn, strategy retry.Strategy, db *BaseDB) *BaseConn {
 	if strategy == nil {
 		strategy = &retry.FiniteRetryStrategy{}
 	}
-	return &BaseConn{conn, strategy}
+	return &BaseConn{conn, strategy, db}
+}
+
+// Reset resets this BaseConn, close old one and generate new *sql.Conn from its BaseDB
+func (conn *BaseConn) Reset(tctx *tcontext.Context) error {
+	err := conn.ParentDB.CloseBaseConn(conn)
+	if err != nil {
+		return err
```

Comment:
should we need to ignore this closing error?

Reply:
How about print warning log?

Reply:
add warn log [here](https://github.com/pingcap/dm/pull/266/files#diff-e25c1def89789065c4904c1d5adc0867R84), and then ignore it, Normally this couldn't happen